File: Backend/src/main/java/lk/fleet/service/VehicleAccidentService.java
package lk.fleet.service;

import lk.fleet.dto.VehicleAccidentDTO;
import lk.fleet.entity.DriverVehicle;
import lk.fleet.entity.VehicleAccident;

import java.util.List;

public interface VehicleAccidentService {

    VehicleAccidentDTO addAccident(VehicleAccident vehicleAccident);
//   DriverVehicle addAccident(DriverVehicle vehicleAccident);

    VehicleAccidentDTO updateVehicleAccident (String vehicleAccidentID, VehicleAccident vehicleAccident);

    boolean deleteAccident (String vehicleAccidentID);

    List<VehicleAccidentDTO> getVehicleAccidents();

    boolean chkVehicle(String vehicleId);

    boolean chkDriver(String vehicleId,String driverId);
}

File: Frontend/src/app/_service/vehicle-accident.service.ts
import {Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class VehicleAccidentService {

  accident;

  constructor(private http: HttpClient) {
  }

  addAccident(accidentDetail): Observable<any> {
    return this.http.post<any>(environment.backend_url + "/accident/addVehicleAccident", accidentDetail);
  }

  getVehicleAccidents(): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/accident/getVehicleAccidents");
  }

  updateVehicleAccident(accidentDetail): Observable<any> {
    return this.http.put<any>(environment.backend_url + "/accident/updateVehicleAccident/" + accidentDetail.vehicleAccidentID, accidentDetail);
  }

  deleteAccident(vehicleAccidentID): Observable<any> {
    return this.http.delete<any>(environment.backend_url + "/accident/deleteVehicleAccident/" + vehicleAccidentID);
  }

  getAccidentById(vehicleAccidentID): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/accident/getAccidentById/" + vehicleAccidentID);
  }

  chkVehicle(vehicleId): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/accident/chkVehicle/" + vehicleId);
  }

  chkDriver(vehicleId, driverId): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/accident/chkDriver/" + vehicleId + '/' + driverId);
  }


}
